numpy/core/src/multiarray/temp_elide.c

```diff
@@ -359,7 +359,7 @@ can_elide_temp_unary(PyArrayObject * m1)
 {
     int cannot;
     if (Py_REFCNT(m1) != 1 || !PyArray_CheckExact(m1) ||
-            PyArray_DESCR(m1)->type_num == NPY_VOID ||
+            PyArray_DESCR(m1)->type_num >= NPY_OBJECT ||
```

Comment:
Should probably fix the release notes to say that elision is added for builtin _numeric_ types